### DISCUSSION TITLE
Should we implement a "setFintroller" function in the HToken?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [hifi-finance/hifi#83] [Ideas]
Since [v1.8.0](https://github.com/hifi-finance/hifi/tree/%40hifi/protocol%401.8.0), the `HToken` contract expects a new argument in its constructor, which is the address of the Fintroller contract.

Would it be worth it to add a `setFintroller` function, callable only by the owner of the contract?

There's already a `setBalanceSheet` function .. though that hasn't been ever needed thus far.

**Comment by scorpion9979** [comment]
I think I would rather we remove `setBalanceSheet` than add `setFintroller`. Doing so would save 22 gas on any external function which has a signature that sorts after the one for `setBalanceSheet` in the function dispatch table. Of course, same result could be accomplished by renaming `setBalanceSheet` in order to push its function signature to the last position in the function dispatch table.

References:
https://medium.com/joyso/solidity-how-does-function-name-affect-gas-consumption-in-smart-contract-47d270d8ac92
https://ethereum.stackexchange.com/questions/68098/why-does-adding-a-view-function-make-write-functions-cost-more-gas

**Reply by PaulRBerg** [reply]
Having `setBalanceSheet` is really just an insurance policy against any potential bugs in the `BalanceSheet` contracts. Supposedly, we would patch those bugs ASAP, upgrade the proxy implementation, then call `setBalanceSheet` on the hToken.

Through this lens, 22 gas is a negligible cost.
